src/TerminalFighter/MenuState/MenuState.cpp

```diff
@@ -31,16 +43,32 @@ gamestates::GameStateName MenuState::run() {
                 case SDL_KEYDOWN:
                     switch (event.key.keysym.sym) {
                         case SDLK_RETURN:
-                            LOG(INFO) << "Enter was Pressed!";
-                            return gamestates::TEST;
+                            LOG(INFO) << "Enter was pressed!";
+                            if (main_menu.get_current_selection() == START) {
+                                LOG(INFO) << "Start selected";
+                                return gamestates::TEST;
+                            } else if (main_menu.get_current_selection() == QUIT) {
+                                LOG(INFO) << "Quit selected";
+                                exit(0);
```

Comment:
Return `gamestates::EXIT` instead so that higher level code can handle a proper exit.